shared/src/main/scala/org/adridadou/openlaw/parser/template/VariableDefinition.scala

```diff
@@ -167,23 +167,45 @@ case class VariableDefinition(name: VariableName, variableTypeDefinition:Option[
       case Parameters(parameterMap) =>
         parameterMap.toMap.get("options").flatMap({
           case OneValueParameter(expr) =>
-            val badType = expr.expressionType(executionResult)
-            if(badType =!= varType(executionResult)) {
-              Some(s"options element error! should be of type ${varType(executionResult).name} but is ${badType.name} instead")
-            } else {
-              None
-            }
+            validateOptions(Seq(expr), executionResult)
           case ListParameter(exprs) =>
-            val exprTypes = exprs
-            exprTypes.find(_.expressionType(executionResult) =!= varType(executionResult))
-            .map(expr => s"options element error! should be of type ${varType(executionResult).name} but ${expr.toString} is ${expr.expressionType(executionResult).name} instead")
+            validateOptions(exprs, executionResult)
           case _ =>
             None
         })
       case _ => None
     })
   }
 
+  private def validateOptions(exprs:Seq[Expression], executionResult: TemplateExecutionResult):Option[String] = {
+    val currentType = varType(executionResult)
+
+    currentType match {
+      case DefinedChoiceType(choices,typeName) =>
+        exprs.flatMap(validateOption(_, choices, typeName, executionResult)).headOption
+      case _ =>
+        val exprTypes = exprs
+        exprTypes.find(_.expressionType(executionResult) =!= currentType)
```

Comment:
Should this just be `exprs.find(........)`?

Reply:
I need the map to have a nice error message.

Reply:
You're right